chains/evm/calls/deploy.go

```diff
@@ -14,82 +15,82 @@ import (
-	parsed, err := abi.JSON(strings.NewReader(consts.ERC721PresetMinterPauserABI))
+func DeployErc721(c client.ClientDeployer, txFabric client.TxFabric, gasPriceClient client.GasPricer, name, symbol, baseURI string) (common.Address, error) {
+	parsed, err := abi.JSON(strings.NewReader(consts.PresetMinterPauserABI))
 	if err != nil {
 		return common.Address{}, err
 	}
-	address, err := deployContract(c, parsed, common.FromHex(consts.ERC721PresetMinterPauserBin), txFabric, gasPriceClient, name, symbol, baseURI)
+	address, err := DeployContract(c, parsed, common.FromHex(consts.PresetMinterPauserBin), txFabric, gasPriceClient, name, symbol, baseURI)
 	if err != nil {
 		return common.Address{}, err
 	}
 	return address, nil
 }
 
-func DeployErc721Handler(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, bridgeAddress common.Address) (common.Address, error) {
+func DeployErc721Handler(c client.ClientDeployer, txFabric client.TxFabric, gasPriceClient client.GasPricer, bridgeAddress common.Address) (common.Address, error) {
 	log.Debug().Msgf("Deploying ERC721 Handler with params: %s", bridgeAddress.String())
-	parsed, err := abi.JSON(strings.NewReader(consts.ERC721HandlerABI))
+	parsed, err := abi.JSON(strings.NewReader(consts.HandlerABI))
 	if err != nil {
 		return common.Address{}, err
 	}
-	address, err := deployContract(c, parsed, common.FromHex(consts.ERC721HandlerBin), txFabric, gasPriceClient, bridgeAddress)
+	address, err := DeployContract(c, parsed, common.FromHex(consts.HandlerBin), txFabric, gasPriceClient, bridgeAddress)
 	if err != nil {
 		return common.Address{}, err
 	}
 	return address, nil
 }
 
-func DeployGenericHandler(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, bridgeAddress common.Address) (common.Address, error) {
+func DeployGenericHandler(c client.ClientDeployer, txFabric client.TxFabric, gasPriceClient client.GasPricer, bridgeAddress common.Address) (common.Address, error) {
 	log.Debug().Msgf("Deploying Generic Handler with params: %s", bridgeAddress.String())
 	parsed, err := abi.JSON(strings.NewReader(consts.GenericHandlerABI))
 	if err != nil {
 		return common.Address{}, err
 	}
-	address, err := deployContract(c, parsed, common.FromHex(consts.GenericHandlerBin), txFabric, gasPriceClient, bridgeAddress)
+	address, err := DeployContract(c, parsed, common.FromHex(consts.GenericHandlerBin), txFabric, gasPriceClient, bridgeAddress)
 	if err != nil {
 		return common.Address{}, err
 	}
 	return address, nil
 }
 
-func deployContract(client ClientDeployer, abi abi.ABI, bytecode []byte, txFabric TxFabric, gasPriceClient GasPricer, params ...interface{}) (common.Address, error) {
+func DeployContract(client client.ClientDeployer, abi abi.ABI, bytecode []byte, txFabric client.TxFabric, gasPriceClient client.GasPricer, params ...interface{}) (common.Address, error) {
```

Comment:
This function is basically a Transactor.Transact method, so maybe in order to make it testable we we should change  DeployGenericHandler signature so it accepts Transactor  as argument?

Reply:
Or you just have not touched it because was focusing only on bridge?